L1Trigger/Phase2L1ParticleFlow/interface/ParametricResolution.h

```diff
@@ -0,0 +1,67 @@
+#ifndef L1Trigger_Phase2L1ParticleFlow_ParametricResolution_h
+#define L1Trigger_Phase2L1ParticleFlow_ParametricResolution_h
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/Exception.h"
+#include <vector>
+#include <cmath>
+
+namespace l1tpf {
+  class ParametricResolution {
+  public:
+    ParametricResolution() {}
+    ParametricResolution(const edm::ParameterSet &cpset) {
+      std::vector<double> etaBins = cpset.getParameter<std::vector<double>>("etaBins");
+      std::vector<double> offset = cpset.getParameter<std::vector<double>>("offset");
+      std::vector<double> scale = cpset.getParameter<std::vector<double>>("scale");
+      etas.insert(etas.end(), etaBins.begin(), etaBins.end());
+      scales.insert(scales.end(), scale.begin(), scale.end());
+      offsets.insert(offsets.end(), offset.begin(), offset.end());
+      if (cpset.existsAs<std::vector<double>>("ptMin")) {
```

Comment:
as I don't know all possible places where this class will be created, I would prefer not to rely on the paramater set validation for this

Reply:
@makortel @Dr15Jones please comment
I think it's possible to use `fillPSetDescription()` ([twiki](https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideConfigurationValidationAndHelp)) to help automate this

Reply:
By quick look it indeed seems that the pattern of `ParametricResolution::fillPSetDescription()` being called from an EDModule `fillDescriptions()` would work here. That would be much preferred over `existsAs()`.

On the other hand, if this parameter is supposed to be either `std::vector<double>` or `double`, I don't see how `existsAs()` could be avoided even with `fillDescriptions()`.

Reply:
I think this could follow the suggestion I made in another PR: if the vector has length 1, then fill the vector with the 1 value (replacing the `double` case)

Reply:
The advantage of the `fillDescriptions` is it will require that the parameter named `ptMin` exists and it is either a `std::vector<double>` or a `double`. The present code has a problem if someone mistypes the parameter name, say `pt_min` or `ptmin`. Then in this logic the value of 0 will be used without realizing a problem has happened.

`fillDescriptions` can handle the case where a parameter is allowed to be of different types and in that case you do have to use `existsAs` when requesting the data.